core/test/financial-templates/PrecisionErrors.js

```diff
@@ -0,0 +1,290 @@
+    let actualFeesCollected; // Delta between starting store balance and ending store balance.
+
+    /**
+     * @notice SETUP THE TEST
+     */
+    // 1) Create position.
+    await collateral.approve(pricelessPositionManager.address, startCollateralAmount.toString(), { from: sponsor });
+    await pricelessPositionManager.create(
+      { rawValue: startCollateralAmount.toString() },
+      { rawValue: startTokenAmount.toString() },
+      { from: sponsor }
+    );
+    // 2) Set fee rate per second.
+    await store.setFixedOracleFeePerSecond({ rawValue: toWei(feeRatePerSecond) });
+    // 3) Move time in the contract forward by 1 second to capture unit fee.
+    const startTime = await pricelessPositionManager.getCurrentTime();
+    await pricelessPositionManager.setCurrentTime(startTime.addn(1));
+
+    /**
+     * @notice TEST INVARIANTS
+     */
+    const startingStoreBalance = await collateral.balanceOf(store.address);
+    adjustedCollateralAmount = await pricelessPositionManager.getCollateral(sponsor);
+    actualCollateralAmount = await collateral.balanceOf(pricelessPositionManager.address);
+
+    // Test 1) The adjusted and actual collateral amount is the same to start, pre-fees.
+    assert.equal(adjustedCollateralAmount.toString(), actualCollateralAmount.toString());
+    console.group("** Pre-Fees: **");
+    console.log("- Store Collateral:", startingStoreBalance.toString());
+    console.log("- Adjusted Collateral (returned by contract.getCollateral()):", adjustedCollateralAmount.toString());
+    console.log("- Actual Collateral (returned by token.balanceOf()):", actualCollateralAmount.toString());
+    console.groupEnd();
+
+    /**
+     * @notice RUN THE TEST ONCE AND PRODUCE KNOWN DRIFT
+     */
+    await pricelessPositionManager.payFees();
+    endingStoreBalance = await collateral.balanceOf(store.address);
+    actualFeesCollected = endingStoreBalance.sub(startingStoreBalance).toString();
+    adjustedCollateralAmount = await pricelessPositionManager.getCollateral(sponsor);
+    actualCollateralAmount = await collateral.balanceOf(pricelessPositionManager.address);
+    drift = actualCollateralAmount.sub(toBN(adjustedCollateralAmount.rawValue));
+
+    // Test 1) The correct fees are paid are regardless of precision loss.
+    assert.equal(expectedFeesCollectedPerPeriod.toString(), actualFeesCollected.toString());
+
+    // Test 2) Due to the precision error mentioned above, `getCollateral()` should return
+    // slightly less than what we are expecting.
+    assert(drift.gt(toBN(0)));
+    console.group("** After 1 second: **");
+    console.log("- Fees collected: ", actualFeesCollected.toString());
+    console.log("- Sponsor's credited collateral returned by getCollateral(): ", adjustedCollateralAmount.toString());
+    console.log("- Collateral owned by contract:", actualCollateralAmount.toString());
+    console.log("- Drift:", parseFloat(drift.toString()) / 1e18);
```

Comment:
Does this cause any loss in how it prints the precision due to the arithmetic you're doing here?

Reply:
I haven't seen any loss in the drift but I actually reverted this and just printed out the raw drift